### ISSUE TITLE
ApplicationRunner should not auto_reconnect after failed login

### ISSUE THREAD

**Issue by Mahedros** [crossbario/autobahn-python#878]
When auto_reconnect is enabled, the ApplicationRunner does not know whether or not it ever successfully connected in the first place. As a result, connections lost due to incorrect login information are constantly re-attempted with the same incorrect information.

A check that prevents this exists in Autobahn JS: 
https://github.com/crossbario/autobahn-js/blob/master/lib/connection.js#L349

**Comment by oberstet** [comment]
Yep. And this case should also be signaled using a dedicated error URI / reason.

Eg we probably want that in https://github.com/crossbario/autobahn-python/blob/master/autobahn/wamp/exception.py
